созданы РО и локаторы, классы элементов страниц

iasem84 opened the pull request.

iasem84 committed:
созданы РО и локаторы, классы элементов страниц

iasem84 committed:
готовый проект Диплом 3

iasem84 committed:
add allure report

okryukova commented on src/main/java/practicum/Constants.java:
```diff
@@ -0,0 +1,31 @@
+package practicum;
+
+public class Constants {
+    public static final String INPUT_EMAIL_FIELD_LOCATOR = ".//label[contains(text(), 'Email')]/following::input[1]";
```
⛔️Нужно исправить.  Локаторы нужно хранить в page object. Перенеси их, пожалуйста, в нужные классы и Constants

okryukova commented on src/main/java/practicum/api/User.java:
```diff
@@ -0,0 +1,72 @@
+
+    @Step("Create user")
+    public void createUser() {
+        json = "{\"email\": \"" + email + "\", \"password\": \"" + password + "\", \"name\": \"" + name + "\" }";
```
⛔️Нужно исправить.  Не нужно формировать в ручную тело запроса. Прочитай, плиз, внимательно теорию, раздел Сериализация (Спринт 7/9 → Тема 2/4: Java. API → Урок 3/8). Ты подключила библиотеку Gson к проекту -- она сама преобразует объект в json там, где это необходимо (при передачи json-данных в боди запроса).
Поправь, пожалуйста, этот момент во всем коде

okryukova commented on src/main/java/practicum/api/User.java:
```diff
@@ -0,0 +1,72 @@
+    private String token;
+
+    public User() {
+        Faker faker = new Faker();
```
⚠️Можно улучшить.  Генерация юзера не связана с апи запросами, целесообразно перенести ее в отдельный класс

okryukova commented on src/main/java/practicum/api/User.java:
```diff
@@ -0,0 +1,72 @@
+
+public class User extends RestApi {
+
+    private final String email;
```
⛔️Нужно исправить. Данные и апи методы нужно разнести в разные классы.

okryukova commented on src/main/java/practicum/pageobject/MainPage.java:
```diff
@@ -0,0 +1,52 @@
+    public void bunsButtonClick() {
+        ButtonElement bunsButton = new ButtonElement(BUNS_BUTTON_LOCATOR);
+        bunsButton.clickButton();
+        Configuration.timeout = 3000;
```
⚠️Можно улучшить.  Configuration.timeout задан в BaseUITest. Есть ли смысл менять его в каждом тесте?

okryukova commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,43 @@
+        loginPage.login(user.getEmail(), user.getPassword());
+
+        mainPage.personalAccountButtonClick();
+        assertEquals(PERSONAL_ACCOUNT_PAGE_URL, driver().url());
```
⛔️Нужно исправить. Проверь, пожалуйста, ассерт, тест упал

org.junit.ComparisonFailure: expected:<...parties.site/account[/profile]> but was:<...parties.site/account[]>
	at LoginTest.userLoginTest(LoginTest.java:41)

iasem84 committed:
1. Локаторы перенесены в PO
2. Автоматическая генерация тела запроса
3. Генерация юзера перенсена в класс User
4. Данные и Api методы разнесены в классы User и UserApi соответственно
5. Добавлены ожидания нужной кондиции в тестах конструктора
6. Изменен ассерт в тесте LoginTest - вместо url проверяем отображение кнопки.

iasem84 committed:
1. Добавлено условие в MainPage для стабильности тестов конструктора
